homeassistant/components/tractive/__init__.py

```diff
@@ -0,0 +1,153 @@
+"""The tractive integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+import aiotractive
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    DOMAIN,
+    RECONNECT_INTERVAL,
+    SERVER_UNAVAILABLE,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+    TRACKER_POSITION_UPDATED,
+)
+
+PLATFORMS = ["device_tracker"]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tractive from a config entry."""
+    data = entry.data
+
+    hass.data.setdefault(DOMAIN, {})
+
+    client = aiotractive.Tractive(
+        data[CONF_EMAIL], data[CONF_PASSWORD], session=async_get_clientsession(hass)
+    )
+    try:
+        creds = await client.authenticate()
+    except aiotractive.exceptions.TractiveError as error:
+        await client.close()
+        raise ConfigEntryNotReady from error
```

Comment:
If it's an authentication error we should probably raise `ConfigEntryAuthFailed` and add a reauth step in the config flow, to let the user re-authenticate.

For other connection errors, that we think may be resolved automatically, we raise `ConfigEntryNotReady`.